homeassistant/components/axis/config_flow.py

```diff
@@ -137,36 +138,56 @@ async def _create_entry(self):
         title = f"{model} - {self.device_config[CONF_MAC]}"
         return self.async_create_entry(title=title, data=self.device_config)
 
-    async def async_step_zeroconf(self, discovery_info):
+    async def async_step_dhcp(self, discovery_info: dict):
+        """Prepare configuration for a DHCP discovered Axis device."""
+        return await self._process_discovered_device(
+            {
+                CONF_HOST: discovery_info[IP_ADDRESS],
+                CONF_MAC: discovery_info.get(MAC_ADDRESS),
+                CONF_NAME: discovery_info.get(HOSTNAME),
+                CONF_PORT: DEFAULT_PORT,
+            }
+        )
+
+    async def async_step_zeroconf(self, discovery_info: dict):
         """Prepare configuration for a discovered Axis device."""
-        serial_number = discovery_info["properties"]["macaddress"]
+        return await self._process_discovered_device(
+            {
+                CONF_HOST: discovery_info[CONF_HOST],
+                CONF_MAC: discovery_info["properties"]["macaddress"],
+                CONF_NAME: discovery_info["hostname"][:-7],
+                CONF_PORT: discovery_info[CONF_PORT],
+            }
+        )
 
-        if serial_number[:6] not in AXIS_OUI:
+    async def _process_discovered_device(self, device: dict):
+        """Prepare configuration for a discovered Axis device."""
+        if device[CONF_MAC][:6] not in AXIS_OUI:
             return self.async_abort(reason="not_axis_device")
 
-        if is_link_local(ip_address(discovery_info[CONF_HOST])):
+        if is_link_local(ip_address(device[CONF_HOST])):
             return self.async_abort(reason="link_local_address")
 
-        await self.async_set_unique_id(serial_number)
+        await self.async_set_unique_id(device[CONF_MAC])
```

Comment:
Is the mac address formatting going to be the same between sources?

`from homeassistant.helpers.device_registry import format_mac`.. but that doesn't cover existing ones

Reply:
I will verify this between all sources

Reply:
Standardised MAC according to format_mac in separate PR https://github.com/home-assistant/core/issues/37372